Android notification service crashes

```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: de.tutao.tutanota:pushprocess, PID: 18797
    java.lang.RuntimeException: Unable to create service de.tutao.tutanota.push.PushNotificationService: java.lang.NullPointerException
        at android.app.ActivityThread.handleCreateService(ActivityThread.java:4199)
        at android.app.ActivityThread.access$1500(ActivityThread.java:238)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1933)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7705)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:952)
     Caused by: java.lang.NullPointerException
        at de.tutao.tutanota.alarms.EncryptedRepeatRuleKt.decrypt(EncryptedRepeatRule.kt:28)
        at de.tutao.tutanota.alarms.AlarmNotificationEntityKt.decrypt(AlarmNotificationEntity.kt:129)
        at de.tutao.tutanota.alarms.AlarmNotificationsManager.reScheduleAlarms(AlarmNotificationsManager.kt:27)
        at de.tutao.tutanota.push.PushNotificationService.onCreate(PushNotificationService.kt:33)
        at android.app.ActivityThread.handleCreateService(ActivityThread.java:4187)
        at android.app.ActivityThread.access$1500(ActivityThread.java:238) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1933) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:223) 
        at android.app.ActivityThread.main(ActivityThread.java:7705) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(R
```

This is line 28
https://github.com/tutao/tutanota/blob/727cefb30169a60f9f7a35d15f02015695d3623d/app-android/app/src/main/java/de/tutao/tutanota/alarms/EncryptedRepeatRule.kt#L28

endValue is not always there?

--- a/app-android/app/src/main/java/de/tutao/tutanota/alarms/AlarmModel.kt
+++ b/app-android/app/src/main/java/de/tutao/tutanota/alarms/AlarmModel.kt
@@ -14,7 +14,7 @@ object AlarmModel {
 			frequency: RepeatPeriod,
 			interval: Int,
 			endType: EndType?,
-			endValue: Long,
+			endValue: Long?,
 			alarmTrigger: AlarmTrigger,
 			localTimeZone: TimeZone,
 			callback: AlarmIterationCallback,
@@ -28,9 +28,9 @@ object AlarmModel {
 
 		val endDate = if (endType == EndType.UNTIL) {
 			if (isAllDayEvent) {
-				getAllDayDateLocal(Date(endValue), localTimeZone)
+				getAllDayDateLocal(Date(endValue!!), localTimeZone)
 			} else {
-				Date(endValue)
+				Date(endValue!!)
 			}
 		} else {
 			null
@@ -38,9 +38,9 @@ object AlarmModel {
 		val calendar = Calendar.getInstance(if (isAllDayEvent) localTimeZone else timeZone)
 		var occurrences = 0
 		var futureOccurrences = 0
-		while (futureOccurrences < OCCURRENCES_SCHEDULED_AHEAD
-				&& (endType != EndType.COUNT
-						|| occurrences < endValue)
+		while (
+				futureOccurrences < OCCURRENCES_SCHEDULED_AHEAD &&
+				(endType != EndType.COUNT || occurrences < endValue!!)
 		) {
 			calendar.time = calcEventStart
 			incrementByRepeatPeriod(calendar, frequency, interval * occurrences)

--- a/app-android/app/src/main/java/de/tutao/tutanota/alarms/AlarmNotificationEntity.kt
+++ b/app-android/app/src/main/java/de/tutao/tutanota/alarms/AlarmNotificationEntity.kt
@@ -13,7 +13,7 @@ class RepeatRule(
 		val frequency: RepeatPeriod,
 		val interval: Int,
 		val timeZone: TimeZone,
-		val endValue: Long,
+		val endValue: Long?,
 		val endType: EndType,
 )
 

--- a/app-android/app/src/main/java/de/tutao/tutanota/alarms/EncryptedRepeatRule.kt
+++ b/app-android/app/src/main/java/de/tutao/tutanota/alarms/EncryptedRepeatRule.kt
@@ -17,15 +17,15 @@ class EncryptedRepeatRule(
 
 fun EncryptedRepeatRule.decrypt(crypto: AndroidNativeCryptoFacade, sessionKey: ByteArray): RepeatRule {
 	val repeatPeriodNumber = crypto.decryptNumber(frequency, sessionKey)
-	val repeatPeriod = RepeatPeriod.get(repeatPeriodNumber)
+	val repeatPeriod = RepeatPeriod[repeatPeriodNumber]
 
 	val endTypeNumber = crypto.decryptNumber(endType, sessionKey)
-	val endType = EndType.get(endTypeNumber)
+	val endType = EndType[endTypeNumber]
 	return RepeatRule(
 			frequency = repeatPeriod,
 			interval = crypto.decryptNumber(interval, sessionKey).toInt(),
 			timeZone = TimeZone.getTimeZone(crypto.decryptString(timeZone, sessionKey)),
-			endValue = crypto.decryptNumber(endValue!!, sessionKey),
+			endValue = if (endValue != null) crypto.decryptNumber(endValue, sessionKey) else null,
 			endType = endType,
 	)
 }

